src/test/py/bazel/launcher_test.py

```diff
@@ -564,6 +564,75 @@ def testWindowsNativeLauncherInNonEnglishPath(self):
     self.AssertExitCode(exit_code, 0, stderr)
     self.assertEqual('helloworld', ''.join(stdout))
 
+  def testWindowsNativeLauncherInLongPath(self):
+    if not self.IsWindows():
+      return
+    self.ScratchFile('WORKSPACE')
+    self.ScratchFile('bin/BUILD', [
+        'java_binary(',
+        '  name = "bin_java",',
+        '  srcs = ["Main.java"],',
+        '  main_class = "Main",',
+        ')',
+        'sh_binary(',
+        '  name = "bin_sh",',
+        '  srcs = ["main.sh"],',
+        ')',
+        'py_binary(',
+        '  name = "bin_py",',
+        '  srcs = ["bin_py.py"],',
+        ')',
+    ])
+    self.ScratchFile('bin/Main.java', [
+        'public class Main {',
+        '  public static void main(String[] args) {'
+        '    System.out.println("helloworld");',
+        '  }',
+        '}',
+    ])
+    self.ScratchFile('bin/main.sh', [
+        'echo "helloworld"',
+    ])
+    self.ScratchFile('bin/bin_py.py', [
+        'print("helloworld")',
+    ])
+
+    exit_code, stdout, stderr = self.RunBazel(['info', 'bazel-bin'])
+    self.AssertExitCode(exit_code, 0, stderr)
+    bazel_bin = stdout[0]
+
+    exit_code, _, stderr = self.RunBazel(['build', '//bin/...'])
+    self.AssertExitCode(exit_code, 0, stderr)
+
+    # Create a directory with a path longer than 260
+    long_dir_path = "./" + "/".join(["a" * 100, "b" * 100, "c" * 100])
```

Comment:
This could be shortened with the `AAAAAA~1\BBBBBB~1\CCCCCC~1` style.
Could you create a non-shortenable path? (I can recommend names like `aaaaaaaa.aaa` -- that is exactly 8.3 characters long and cannot be shortened.)

Reply:
Done